node/core/approval-voting/src/approval_checking.rs

```diff
@@ -938,6 +948,48 @@ mod tests {
 			},
 		);
 	}
+
+	#[test]
+	fn filled_tranche_iterator_yields_sequential_tranches() {
+		const PREFIX: u32 = 10;
+
+		let test_tranches = vec![
+			vec![],	 // empty set
+			vec![0],	// zero start
+			vec![0, 3], // zero start with gap
+			vec![2],	// non-zero start
+			vec![2, 4], // non-zero start with gap
+		];
```

Comment:
Some test cases with longer gaps and longer sequential runs would be good

Reply:
Yes, the goal here was to get started with something and see whether it's moving in the right direction. I will continue to expand on this.